### PR TITLE
[Hub Generated] Review request for Microsoft.AzureStack to add version stable/2016-01-01

### PR DESCRIPTION
**Pull request by ionuttulai**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/ionuttulai/azure-rest-api-specs/dev-azurestack-Microsoft.AzureStack-2016-01-01...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ionuttulai** [comment]
This PR is adding missing api version 2016-01-01 for swagger completeness.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/stable/2016-01-01/AzureStack.json [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Resource": {
```
Comment (varunkumta) [inline specification/azurestack/resource-manager/Microsoft.AzureStack/stable/2016-01-01/AzureStack.json:56]: >Resource [](start = 5, length = 8)

you don't need to define your own - you can reuse from common-types/resource-management/v2/types.json

Comment (varunkumta) [reply specification/azurestack/resource-manager/Microsoft.AzureStack/stable/2016-01-01/AzureStack.json:56]: and pretty much everything else in this file

---
In reply to: [572604325](https://github.com/Azure/azure-rest-api-specs/pull/12876#discussion_r572604325) [](ancestors = 572604325)

Comment (ionuttulai) [reply specification/azurestack/resource-manager/Microsoft.AzureStack/stable/2016-01-01/AzureStack.json]: I removed the generic types and referenced the ones in common-types folder.

**Comment by varunkumta** [comment]
> This PR is adding missing api version 2016-01-01 for swagger completeness.

I have updated the PR checklist accordingly

**Comment by ionuttulai** [comment]
@varunkumta we are taking care of s360 alerts but not by changes to an existing API version, but by adding a missing one for swagger completeness - that's why I initially had it marked as "new API version" .

Here's a link to the s360 issues we are addressing with this PR

https://vnext.s360.msftcloudes.com/blades/unifiedplatform?blade=AssignedTo:All~KPI:6dc4e485-2d1b-449c-a224-30b183391437~SLA:1~Forums:All~_loc:UnifiedPlatform&def=&peopleBasedNodes=avishwan_team;maximpop_team&tile=S360_ServiceId:9537bbc3-8945-4a80-a398-9e894170cc28~_loc:__key__UnifiedPlatform__6dc4e485-2d1b-449c-a224-30b183391437&global=4:9537bbc3-8945-4a80-a398-9e894170cc28 
 
Cc: @markcowl for awareness

**Comment by varunkumta** [comment]
> @varunkumta we are taking care of s360 alerts but not by changes to an existing API version, but by adding a missing one for swagger completeness - that's why I initially had it marked as "new API version" .
> 
> Here's a link to the s360 issues we are addressing with this PR
> 
> https://vnext.s360.msftcloudes.com/blades/unifiedplatform?blade=AssignedTo:All~KPI:6dc4e485-2d1b-449c-a224-30b183391437~SLA:1~Forums:All~_loc:UnifiedPlatform&def=&peopleBasedNodes=avishwan_team;maximpop_team&tile=S360_ServiceId:9537bbc3-8945-4a80-a398-9e894170cc28~_loc:__key__UnifiedPlatform__6dc4e485-2d1b-449c-a224-30b183391437&global=4:9537bbc3-8945-4a80-a398-9e894170cc28
> 
> Cc: @markcowl for awareness

yes, pretty sure here existing means 'publicly available' not 'documented in swagger' - as I see a lot of traffic in ARM for your RP and this apiversion

**Comment by ionuttulai** [comment]
> > @varunkumta we are taking care of s360 alerts but not by changes to an existing API version, but by adding a missing one for swagger completeness - that's why I initially had it marked as "new API version" .
> > Here's a link to the s360 issues we are addressing with this PR
> > https://vnext.s360.msftcloudes.com/blades/unifiedplatform?blade=AssignedTo:All~KPI:6dc4e485-2d1b-449c-a224-30b183391437~SLA:1~Forums:All~_loc:UnifiedPlatform&def=&peopleBasedNodes=avishwan_team;maximpop_team&tile=S360_ServiceId:9537bbc3-8945-4a80-a398-9e894170cc28~_loc:__key__UnifiedPlatform__6dc4e485-2d1b-449c-a224-30b183391437&global=4:9537bbc3-8945-4a80-a398-9e894170cc28
> > Cc: @markcowl for awareness
> 
> yes, pretty sure here _existing_ means 'already publicly available' not 'existing in swagger' - as I see a lot of traffic in ARM for your RP and this apiversion

Sounds good. Thanks for the review!

**Review by markcowl — APPROVED** [review]

**Comment by markcowl** [comment]
@ionuttulai We cna merge when the ci jobs have passed

**Comment by ionuttulai** [comment]
> @ionuttulai We cna merge when the ci jobs have passed

The pipeline has been running for almost 24 hours now. Is that expected, or is there a way to re-queue it?

**Comment by ionuttulai** [comment]
> @ionuttulai We cna merge when the ci jobs have passed

@markcowl It looks like the pipeline ran successfully but the results were not reported back to the PR. 
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=730156&view=results

**Comment by ionuttulai** [comment]
@markcowl is there anything we can do to re-run the pipeline for the mandatory checks? It looks like it ran successfully but the results never made it back to the PR.

**Comment by markcowl** [comment]
@ionuttulai Yes, we are looking at why the notifications did not make it back here

**Comment by markcowl** [comment]
very latest (passing) run is here: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=736836&view=results

**Pull request merged**